tests/testutil/service.go

```diff
@@ -302,6 +311,46 @@ func (ts *TestService) GetKubeCurrentWeight() int {
 	return getRevisionWeight(app, ts.Version)
 }
 
+// PodsResponsesMatchAvailableReplicas does a GetURL in the App endpoint and stores the response in a slice
+// the length of the resulting slice should represent the number of responsive pods in a service.
+// Returns true if the number of replicas is equal to the length of the responses slice.
+func (ts *TestService) PodsResponsesMatchAvailableReplicas(path string, numberOfReplicas int) bool {
+	i := 0
+	replicasTimesRequests := numberOfReplicas * 5
+	responses := make([]string, 0)
+	for i < replicasTimesRequests {
+		response, err := WaitForURLResponse(ts.GetAppEndpointURL() + path)
+		if err != nil {
+			ts.T.Fatal(err.Error())
+		}
+		i++
+		if !stringInSlice(response, responses) {
+			responses = append(responses, response)
+		}
+	}
+	return len(responses) == numberOfReplicas
+}
+
+// KubeCompareReplicasValues get the app number of ready replicasets with a clientset
+// and returns true if that value match the scale given
+func (ts *TestService) KubeCompareReplicasValues(scale int) bool {
+	err := ts.waitForAvailableReplicas(scale)
+	if err != nil {
+		ts.T.Fatal(err.Error())
+	}
+	clientset := GetClientSet()
```

Comment:
@cbron I've used a clientset here to get the replicas from k8s and compare it with the replica count reported by rio. 
For future test changes we might need to take some time and refactor the kubectl helper methods into their clientset alternatives. 
That would require time though (mostly figure out the proper clientset calls) and we can incrementally work on that on future PRs. Or work on a large change and refactor all of them at once.

Reply:
I like using the clientset. I think in a future PR, we should update our current kubectl commands to use it.
For now, can we update the function name `GetClientSet()` to be more clear what the client set is: `GetKubeClient()` ? or something like that.

Also for consistency, I'd recommend changing this to return the available replicas, like the counterpart of `GetAvailableReplicas()`. This would be similar to how `GetAppEndpointURL()` and `GetKubeAppEndpointURL()` are.